src/Behat/Gherkin/Loader/CucumberNDJsonAstLoader.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+namespace Behat\Gherkin\Loader;
+
+use Behat\Gherkin\Node\BackgroundNode;
+use Behat\Gherkin\Node\ExampleNode;
+use Behat\Gherkin\Node\ExampleTableNode;
+use Behat\Gherkin\Node\FeatureNode;
+use Behat\Gherkin\Node\OutlineNode;
+use Behat\Gherkin\Node\ScenarioInterface;
+use Behat\Gherkin\Node\ScenarioNode;
+use Behat\Gherkin\Node\StepNode;
+use Behat\Gherkin\Node\TableNode;
+
+/**
+ * Loads a feature from cucumber's protobuf JSON format
+ */
+class CucumberNDJsonAstLoader implements LoaderInterface
+{
+
+    public function supports($resource)
+    {
+        return is_string($resource);
+    }
+
+    public function load($resource)
+    {
+        $json = \json_decode(file_get_contents($resource), true);
```

Comment:
this looks wrong to me. NDJson is about having one JSON object per line. you cannot json decode the whole file, otherwise it works only for single-line files.

Reply:
That's true; but currently all the files are single-line! Maybe it's enough to add an assertion that there's only one line?

Reply:
but this file is named NDJsonLoader, not JsonLoader

Reply:
btw, supporting multiple lines would not be an issue: the method already reports an array, so parsing each (non-empty) line with `json_decode` and `self::getFeature` would do the work.

Reply:
That's probably the best approach, thanks

I guess on the test side I can cast from` FeatureNode[]` to `?FeatureNode` (erroring if there's >1) ?

Reply:
Well, for now, yes, because our parser does not support multiple features per parsed file (btw, the description in the phpdoc says it returns a feature array, which is weird in this regard)

Reply:
That's because it's implemented ultimately by things like DirectoryLoader

Reply:
OK now using `file()`